libraries/ms-common/src/event_queue.c

```diff
@@ -4,29 +4,39 @@
 #include <string.h>
 
 #include "event_queue.h"
-#include "pqueue_backed.h"
+#include "fifo.h"
+#include "status.h"
 
 typedef struct EventQueue {
-  PQueueBacked pqueue;
-  PQueueNode queue_nodes[EVENT_QUEUE_SIZE + 1];
-  Event event_nodes[EVENT_QUEUE_SIZE];
+  Fifo fifos[NUM_EVENT_PRIORITIES];
+  Event event_nodes[NUM_EVENT_PRIORITIES][EVENT_QUEUE_SIZE];
```

Comment:
If you wanted to be super space-efficient a stable priority queue would probably allow all the nodes to be shared, but it'd be a lot more complex and probably slower. We have a lot of memory so this probably isn't a big deal.

Reply:
Ack. This seemed like less effort and is more in-keeping with how a lot of other RTOS systems work anyways.